custom_model_runner/datarobot_drum/drum/typeschema_validation.py

```diff
@@ -324,3 +340,56 @@ def revalidate_typeschema(type_schema):
     if "output_requirements" in type_schema:
         for req in type_schema["output_requirements"]:
             req.revalidate(output_validation[req.data["field"]])
+
+
+class SchemaValidator:
+    _input_validator_mapping = {
+        DataTypes.FIELD: DataTypes,
+        SparsityInput.FIELD: SparsityInput,
+        NumColumns.FIELD: NumColumns,
+    }
+    _output_validator_mapping = {
+        DataTypes.FIELD: DataTypes,
+        SparsityOutput.FIELD: SparsityOutput,
+        NumColumns.FIELD: NumColumns,
+    }
+
+    def __init__(self, type_schema, strict=True):
+        self._input_validators = [
+            self._get_validator(schema, self._input_validator_mapping)
+            for schema in type_schema.get("input_requirements", [])
+        ]
+        self._output_validators = [
+            self._get_validator(schema, self._output_validator_mapping)
+            for schema in type_schema.get("output_requirements", [])
+        ]
+        self.strict = strict
+
+    def _get_validator(self, schema, mapping):
+        return mapping[schema["field"]](schema["condition"], schema["value"])
+
+    def validate_inputs(self, dataframe):
+        return self._run_validate(dataframe, self._input_validators, "input")
+
+    def validate_outputs(self, dataframe):
+        return self._run_validate(dataframe, self._output_validators, "output")
+
+    def _run_validate(self, dataframe, validators, step_label):
+        errors = []
+        for validator in validators:
+            errors.extend(validator.validate(dataframe))
+        if len(validators) == 0:
+            print("No type schema for {} provided.".format(step_label))
```

Comment:
Should put non-error prints behind the verbose flag

Reply:
also it would seem that others use logger instead of print